target_link_libraries(test chemfiles) not including headers with CMake

If i try to compile a C code which includes chemfiles.h with the following CMakeLists.txt
```
cmake_minimum_required(VERSION 3.8)
project(test LANGUAGES C)
add_executable(test src/test.c)
find_package(chemfiles 0.9)
target_link_libraries(test PRIVATE chemfiles)
```
I get
```
[ 50%] Building C object CMakeFiles/test.dir/src/test.c.o
test.c:3:10: fatal error: chemfiles.h: No such file or directory
 #include <chemfiles.h>
```
which does not seem right.

Current hack from @Luthaf for me is `target_include_directories(test PRIVATE $ENV{CONDA_PREFIX}/include)`

--- a/.travis.yml
+++ b/.travis.yml
@@ -70,6 +70,7 @@ matrix:
         homebrew:
           packages:
             - gcc@5
+          update: true
     - name: MSVC 64-bit
       os: windows
       env: CMAKE_GENERATOR="Visual Studio 14 2015 Win64"

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,7 +121,7 @@ set_property(TARGET chemfiles PROPERTY SOVERSION ${CHEMFILES_VERSION_SHORT})
 target_include_directories(chemfiles PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
-    $<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}>
+    $<INSTALL_INTERFACE:include>
 )
 
 target_link_libraries(chemfiles ${NETCDF_LIBRARIES} ${ZLIB_LIBRARIES} ${LIBLZMA_LIBRARY})
@@ -143,33 +143,35 @@ generate_export_header(chemfiles
 #----------------------------------------------------------------------------------------#
 # Installation configuration
 #----------------------------------------------------------------------------------------#
-set(INCLUDE_INSTALL_DIR "include")
-install(TARGETS chemfiles EXPORT chemfiles-targets
-    RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib${LIB_SUFFIX}
-    ARCHIVE DESTINATION lib${LIB_SUFFIX}
-)
-install(DIRECTORY include/ DESTINATION include)
-install(DIRECTORY ${PROJECT_BINARY_DIR}/include/ DESTINATION include)
-
 configure_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/cmake/chemfiles-config-version.cmake.in"
     "${CMAKE_CURRENT_BINARY_DIR}/chemfiles-config-version.cmake"
     @ONLY
 )
 
-# Install package files
-set(PACKAGE_LOCATION lib${LIB_SUFFIX}/cmake/chemfiles)
-install(EXPORT chemfiles-targets
-  FILE chemfiles-targets.cmake
-  DESTINATION ${PACKAGE_LOCATION}
+install(TARGETS chemfiles
+    EXPORT chemfiles-targets
+    LIBRARY DESTINATION lib COMPONENT Runtime
+    ARCHIVE DESTINATION lib COMPONENT Development
+    RUNTIME DESTINATION bin COMPONENT Runtime
+    PUBLIC_HEADER DESTINATION include COMPONENT Development
+    BUNDLE DESTINATION bin COMPONENT Runtime
 )
-install(FILES
-    cmake/chemfiles-config.cmake
-    "${CMAKE_CURRENT_BINARY_DIR}/chemfiles-config-version.cmake"
-    DESTINATION ${PACKAGE_LOCATION}
-    COMPONENT Devel
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+    "${PROJECT_SOURCE_DIR}/cmake/chemfiles-config.cmake.in"
+    "${PROJECT_BINARY_DIR}/chemfiles-config.cmake"
+    INSTALL_DESTINATION lib/cmake/chemfiles
 )
+
+install(EXPORT chemfiles-targets DESTINATION lib/cmake/chemfiles)
+install(FILES "${PROJECT_BINARY_DIR}/chemfiles-config-version.cmake"
+              "${PROJECT_BINARY_DIR}/chemfiles-config.cmake"
+        DESTINATION lib/cmake/chemfiles)
+
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ DESTINATION include)
+install(DIRECTORY ${PROJECT_BINARY_DIR}/include/ DESTINATION include)
 
 # uninstall target
 configure_file(

--- a/cmake/chemfiles-config.cmake.in
+++ b/cmake/chemfiles-config.cmake.in
@@ -1,2 +1,4 @@
-# - Config file for the Chemfiles package
+@PACKAGE_INIT@
+
 include("${CMAKE_CURRENT_LIST_DIR}/chemfiles-targets.cmake")
+check_required_components("@PROJECT_NAME@")

